README.md

```diff
@@ -1,2 +1,454 @@
+
+## 5.6 Content Identifiers
+
+UCANs MAY be referenced by content ID (CID), per the [multiformats/cid](https://github.com/multiformats/cid) specification. The resolution of these addresses is left to the implementation and end user, and MAY (non-exclusively) include the following: local store, distributed hash table (DHT), gossip network, or RESTful service.
+
+CIDs MAY be used to refer to any UCAN: a witness in a delegation chain, or an entire UCAN. This has many benefits, some of which are outlined in the implementation recommendations of this document.
+
+Due to the potential for unresolvable CIDs, this SHOULD NOT be the preferred method of transmission. "Inline witnesses" SHOULD be used whenever possible, and complete UCANs SHOULD be expanded. When a CID is used, it is RECOMMENDED that it be substituted as close to the top UCAN as possible (i.e. the invocation), and as few witnesses be referenced by CID, to keep the number of required CID resolutions to a minimum. As UCANs are signed, all further delegations would require CID resolution, and so SHOULD NOT be used when the intention is delegation rather than invocation. 
+
+## 5.7 Revocation
+
+Any issuer of a UCAN MAY later revoke that UCAN, or the capabilities that have been derived from it further downstream in a UCAN chain.
+
+This mechanism is eventually consistent, and SHOULD be considered a last line of defence against abuse. Proactive expiry via time bounds or other constraints SHOULD be preferred, as they do not require learning more information than what would be available on an offline computer.
+
+While some resources are centralized (e.g. access to a server), and others are unbound from specific locations (e.g. a CRDT), and will take longer for the revocation to propagate.
+
+Every resource type SHOULD have a canonical location where its revocations are kept. This list is non-exclusive, and revocation messages MAY be gossiped between peers in a network to share this information more quickly.
+
+It is RECOMMENDED that the canonical revocation store be kept at as close to (or inside) the resource it is about as possible. For example, the WebNative File System maintains a Merkle tree of revoked CIDs at a well-known path. Another example is a centralized server could have an endpoint that lists the revoked UCANs by CID.
+
+Revocations MUST be irreversible. If the revocation was issued in error, a unique UCAN MAY be issued. This prevents confusion as the revocation moves through the network, and makes revocation stores append-only and highly amenable to caching.
```

Comment:
What is the mechanism for issuing a unique UCAN? Would this be adding a nonce?

Reply:
Ah, indeed worth a clarification. The TL;DR is that it's literally any change that will produce a new CID, so a new nonce, changing the time bounds, adding more whitespace, delegating from a different DID, and so on.